src/ck-core/ckfutures.C

```diff
@@ -201,6 +200,51 @@ void *CkWaitFutureID(CkFutureID handle)
   return value;
 }
 
+void Future_dropWaiter(Future* future, const CthThread& th) {
+  auto& waiting = future->waiters;
+  auto it = std::find(waiting.begin(), waiting.end(), th);
+  if (it != waiting.end()) {
+    waiting.erase(it);
+  }
+}
+
+std::pair<void*, CkFutureID> CkWaitAnyID(const std::vector<CkFutureID>& handles) {
+  auto self = CthSelf();
+  auto fs = &(CpvAccess(futurestate));
+  const int n = handles.size();
+  std::vector<Future*> futures;
+  for (auto handle : handles) {
+    auto future = fs->array[handle].get();
+    futures.push_back(future);
+    if (!future->ready) {
+      future->waiters.push_back(self);
+    }
+  }
+  do {
+    for (int i = 0; i < n; i++) {
+      // when a future is ready
+      if (futures[i]->ready) {
+        // drop ourself from all the futures
+        for (auto future: futures) Future_dropWaiter(future, self);
+        // then return the value/handle pair
+        return std::make_pair(futures[i]->value, handles[i]);
+      }
+    }
+    // suspend until a future becomes ready...
+    CthSuspend();
```

Comment:
For some reason, in multi-process configurations, this fails unless `CthSuspend` comes before `future...->value`, even when all the futures are ready (which should be invalid, so things are flipped). Any thoughts on why this may be occurring?